From cd9d7cfa540a7985f7a713affc08bdf1b0f82edf Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Thu, 22 Feb 2018 13:35:42 +0530
Subject: [PATCH 0287/1852] drm: xlnx: scaler: Adding Zero out H-phase array
 before changing resolution

commit 43ea3bfcd867970130caf5a139d8c48064a2e0bc from
https://github.com/Xilinx/linux-xlnx.git

When changing resolutions, the H-phases need to be re-calculated. This
commit zeroes out the H-phase array.

Signed-off-by: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_scaler.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_scaler.c b/drivers/gpu/drm/xlnx/xlnx_scaler.c
index e4c0ec39b41e..0cc0493d72b4 100644
--- a/drivers/gpu/drm/xlnx/xlnx_scaler.c
+++ b/drivers/gpu/drm/xlnx/xlnx_scaler.c
@@ -1522,6 +1522,7 @@ static int xilinx_scaler_bridge_set_input(struct xlnx_bridge *bridge,
 	gpiod_set_value_cansleep(scaler->rst_gpio, XSCALER_RESET_DEASSERT);
 	gpiod_set_value_cansleep(scaler->rst_gpio, XSCALER_RESET_ASSERT);
 	xilinx_scaler_reset(scaler);
+	memset(scaler->H_phases, 0, sizeof(scaler->H_phases));
 
 	xilinx_scaler_write(scaler->base, V_VSCALER_OFF +
 			    XV_VSCALER_CTRL_ADDR_HWREG_HEIGHTIN_DATA, height);
-- 
2.31.1

